gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -46,20 +46,36 @@ var gpii = fluid.registerNamespace("gpii");
                     gpiiKey: "{arguments}.1"
                 },
                 createOnEvent: "onSessionStart"
+            },
+            userLogonRequests: {
+                type: "{arguments}.0",
+                createOnEvent: "onUserLogonRequestReceived",
+                options: {
+                    gpiiKey: "{arguments}.1",
+                    requestPromise: "{arguments}.2"
+                }
             }
         },
         members: {
             sessionIndex: {}, // map of gpiiKey to session component member name, managed by gpii.lifecycleManager.sessionIndexer
-            /* queue for high-level lifecycle manager tasks (start, stop or update).
-             * The entries in the queue are of the format { func: <functionToCall>, arg: <argument> } where
+            /* action queue for high-level lifecycle manager tasks (start, stop or update).
+             * The entries in the action queue are of the format { func: <functionToCall>, arg: <argument> } where
              * <functionToCall> is a single-argument function that returns a promise. The promise should be resolved when
              * the function is complete (including side-effects).
              * A more detailed description: https://github.com/GPII/universal/tree/master/documentation/LifecycleMananger.md
              */
-            queue: [],
-            isProcessingQueue: false // if queue is currently being processed
+            actionQueue: [],
+            userLogonRequestQueue: [],
+            isProcessingActionQueue: false, // if action queue is currently being processed
+            isProcessingUserLogonRequestQueue: false // if user logon queue is currently being processed
         },
         model: {
+            lastNonNoUserLogon: { // Holds information about the last non-noUser login/logout action
```

Comment:
"currentUserLogon" is probably a clearer name. The fact that it only records logons of actual users is ok to leave just in documentation.